Create Monitoring Azure OpenAI docs (#3963)
elastic/observability-docs: docs/en/observability/monitor-azure-openai.asciidoc

. Make note of the storage account name and the storage account access keys because you'll use them later to authenticate your Elastic application's requests to this storage account in <<azure-openai-configure-integration>>.

[discrete]
[[azure-openai-set-up-metrics]]
=== Step 2: Set up Azure to collect metrics

The Azure OpenAI integration metric data stream collects the cognitive service metrics specific to the Azure OpenAI service.
Before {agent} can collect your metrics and send them to {kib}, it needs an app registration to access Azure on your behalf and collect data using the Azure APIs.

Complete the following steps in your Azure instance to register a new Azure app:

. Create the app registration.
. Add credentials to the app.
. Add role assignment to your app.

[discrete]
[[azure-openai-create-app]]
==== Create an app registration

To register your app:

. Go to the https://portal.azure.com/[Azure portal].
. Search for and select **Microsoft Entra ID**.
. Under **Manage**, select **App registrations** → **New registration**.
. Enter a display name for your app (for example, `elastic-agent`).
. Specify who can use the app.

> . The {agent} doesn't use A redirect URI, so you can leave this field blank.

Reviewer: ```suggestion
. The {agent} doesn't use a redirect URI, so you can leave this field blank.
```

. Make note of the **Value** in the **Client secrets** table because you'll use it to specify the **Client Secret** in <<azure-openai-configure-integration>>.
+
WARNING: The secret value is not viewable after you leave this page. Record the value in a safe place.

[discrete]
[[azure-openai-app-role-assignment]]
==== Add role assignment to your app

To add a role assignment to your app:

. From the https://portal.azure.com/[Azure portal], search for and select **Subscriptions**.
. Select the subscription to assign the app.
. Select **Access control (IAM)**.
. Select **Add** → **Add role assignment**.
. In the **Role** tab, search for and select **Monitoring Reader**.
. Click **Next** to open the **Members** tab.
. Select **Assign access to** → **User, group, or service principal**, and select **Select members**.
. Search for and select your app name (for example, "elastic-agent").
. Click **Select**.
. Click **Review + assign**.
. Make note of the **Subscription ID** and **Tenant ID** from your Microsoft Entra because you'll use these to specify settings in the integration.

[discrete]
[[azure-openai-configure-integration]]
=== Step 4: Configure the Azure OpenAI integration

> . Go to the {kib} home page and click **Add integrations**.

Reviewer: ```suggestion
. Go to the {kib} homepage and click **Add integrations**.
```
both are correct, but homepage in one word seems more popular :-)

. In the query bar, search for **Azure OpenAI** and select the Azure OpenAI integration card.
. Click **Add Azure OpenAI**.
. Under Integration settings, configure the integration name and optionally add a description.
+

> TIP: If you don't see options for configuring the integration, you're probably in a workflow designed for new deployments.

Reviewer: ```suggestion
TIP: If you don't have options for configuring the integration, you're probably in a workflow designed for new deployments.
```
accessibility

TIP: Instead of copying the connection string from the RootManageSharedAccessKey policy, you should create a new shared access policy (with permission to listen) and copy the connection string from the new policy.

**Storage account**:: The name of a blob storage account that you set up in <<azure-openai-storage-account-container>>.
You can use the same storage account container for all integrations.

**Storage account key**:: A valid access key defined for the storage account you created in <<azure-openai-storage-account-container>>.
--

[discrete]
[[azure-openai-configure-integration-metrics]]
==== Configure metrics collection

To collect Azure OpenAI metrics, turn on **Collect Azure OpenAI metrics** and specify values for the following required fields:

--
[horizontal]
**Client ID**:: The Application (client) ID that you copied earlier when you created the service principal.

**Client secret**:: The secret value that you copied earlier.

**Tenant ID**:: The tenant ID listed on the main Azure Active Directory Page.

**Subscription ID**:: The subscription ID listed on the main Subscriptions page.
--

After you've finished configuring your integration, click **Save and continue**.

> You'll see a notification that your integration was added. Select **Add {agent} to your hosts**.

Reviewer: ```suggestion
You'll get a notification that your integration was added. Select **Add {agent} to your hosts**.
```

[discrete]
[[azure-openai-install-agent]]
=== Step 5: Install {agent}

IMPORTANT: To get support for the latest API changes from Azure, we recommend

> that using the latest in-service version of {agent} compatible with your

Reviewer: ```suggestion
that you use the latest in-service version of {agent} compatible with your
```
